File: C#/CreationalDesignPattern/CreationalDesignPattern/Refactoring/FactoryMethod/Factory/Dialog.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CreationalDesignPattern.Refactoring.FactoryMethod
{
    public abstract class Dialog
    {
        public void RenderWindow()
        {
            IButton OkButton = CreateButton();
            OkButton.render();
        }

        public abstract IButton CreateButton();
    }
}

File: Java/CreationalDesignPattern/src/main/java/com/ute/refactoring/factorymethod/factory/Dialog.java
package com.ute.refactoring.factorymethod.factory;

import com.ute.refactoring.factorymethod.buttons.Button;

public abstract class Dialog {
    public void renderWindows() {
        Button okButton = createButton();
        okButton.render();
    }

    public abstract Button createButton();
}
